sycl/include/sycl/ext/intel/esimd/detail/simd_obj_impl.hpp

```diff
@@ -124,13 +124,18 @@ constexpr vector_type_t<T, N> make_vector(const T (&&Arr)[N]) {
 }
 
 template <class T, int N, size_t... Is>
-constexpr vector_type_t<T, N> make_vector_impl(T Base, T Stride,
-                                               std::index_sequence<Is...>) {
-  return vector_type_t<T, N>{(T)(Base + ((T)Is) * Stride)...};
+constexpr auto make_vector_impl(T Base, T Stride, std::index_sequence<Is...>) {
+  using CppT = typename element_type_traits<T>::EnclosingCppT;
```

Comment:
I remember you considering optimizing this for low values of N, did that end up not being worth it?

Reply:
I did the initial research for float types and found such tuning worthless.

Just to answer your question here and to show the IR I used `int` type this time, 
and found 2 cases where the old code is 1 instruction faster/shorter:
```
                          old {num math ops : ops}      new {num math ops : ops}
simd<int, 1>           	  0: 				0: 
simd<int, 2>           	* 1: 1xADD			2: 1xADD, 1xMUL
simd<int, 3>		* 3: 2xADD, 1xSHL      		4: 2xADD, 2xMUL (it split 3-elem vec to 2-elem vec + 1-elem vec)
simd<int, 4>		  5: 3xADD, 1xSHL, 1xMUL	2: 1xADD, 1xMUL

simd<float, 1>            0: 				0: 
simd<float, 2>            1: 1xADD			1: 1xMAD
simd<float, 3>            2: 2xADD			2: 2xMAD (3-elem vector ops were split -> 2-elem + 1-elem)
simd<float, 4>            3: 3xADD			1: 1xMAD

```

Reply:
I added few lines of code to tune for integral types and N <= 3: ea002b57eccd2d19adf04196515ddddc0add5f28